Split data-processing-lib/ray into python and ray.

daw3rd opened the pull request:
## Why are these changes needed?
Towards splitting the transforms into separate runtimes, we need to split the library first into python, ray and spark (the latter already exists).

This also fixes some DataAccessS3 bugs introduced with the spark-runtime merge.

## Related issue number (if any).
https://github.com/IBM/data-prep-kit/issues/185

revit13 committed:
Fixes after testing.

Signed-off-by: Revital Sur <eres@il.ibm.com>

daw3rd committed:
Split data-processing-lib/ray into python and ray.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Update workflows for new library and fix spark lib test failure (hopefully)

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix broken build in ingest2parquet and workflows to reference python lib correctly.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Update Dockerfiles to install both python and ray lib source

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix spark transform Dockerfiles to install from data-processing-lib-python

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix failing spark lib test.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix kfp build bug by install python lib in kfp_support_lib venv

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix examples launch.sh script to include both libs in PYTHONPATH

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'runtime-reorg' of github.com:IBM/data-prep-kit into runtime-reorg

revit13 committed:
Fix defaults.image.

Signed-off-by: Revital Sur <eres@il.ibm.com>

daw3rd committed:
Increase timeout for build-images workflow.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix ededup and fdedup to test-image which now requires image dependency.

Signed-off-by: David Wood <dawood@us.ibm.com>

revit13 committed:
revert last change.

Signed-off-by: Revital Sur <eres@il.ibm.com>

daw3rd committed:
Merge branch 'dev' into runtime-reorg

daw3rd committed:
Updated doc overview page to focus on core framework classes w/image.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Rname .make.version TOKENIZER_VERSION to TOKENIZATION_VERSION

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Change building of image to also tag it with quay.io location and not do this during publish

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Change load-image target to have kind load the DOCKER_REMOTE_IMAGE instead of local one.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'dev' into runtime-reorg

daw3rd committed:
Remove fastapi and pillow from data-prep-kit lib dependency

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Merge branch 'runtime-reorg' of github.com:IBM/data-prep-kit into runtime-reorg

daw3rd committed:
Try and generalize the library documentation for the various runtimes.

Signed-off-by: David Wood <dawood@us.ibm.com>

revit13 committed:
Fixes after testing.

Signed-off-by: Revital Sur <eres@il.ibm.com>

revit13 committed:
Remove image_pull_policy from noop wf.

Signed-off-by: Revital Sur <eres@il.ibm.com>

daw3rd committed:
Add debug logging to DataAccessS3/Local and fix initialization bugs in DataAccessS3.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Fix bug in DataAccessS3 get_access_key()

Signed-off-by: David Wood <dawood@us.ibm.com>

blublinsky commented on .make.versions:
```diff
@@ -1,29 +1,31 @@
+PROGLANG_SELECT_VERSION=0.4.0
+CODE_QUALITY_VERSION=0.4.0
+DOC_QUALITY_VERSION=0.4.0
+INGEST_TO_PARQUET_VERSION=0.4.0
```
I think it is about time to align image versions. Assuming that 0.2.0 is the major release, they should be 0.2.x

daw3rd replied:
that would be moving versoins backwards though. definitely not good

blublinsky replied:
Currently they are all random numbers and we have no idea how to link them to the library version. So it is good

daw3rd replied:
I think we agreed this will be taken up in a separate PR

blublinsky commented on .make.versions:
```diff
@@ -1,29 +1,31 @@
-DPK_LIB_VERSION=0.1.1
-DPK_LIB_KFP_VERSION=0.1.1
+DPK_LIB_VERSION=0.2.0
+DPK_LIB_KFP_VERSION=0.2.0
```
Considering that ingest to parquet adds a few changes to the libraries, do you want to postpone PyPI creation? or do this as a minor update?

daw3rd replied:
we can postpone publishing, probably until we're publishing transforms too

blublinsky replied:
That can be a bit tooo long

blublinsky commented on data-processing-lib/python/src/data_processing/data_access/data_access_s3.py:
```diff
@@ -51,19 +51,17 @@ def __init__(
         self.s3_credentials = {} | s3_credentials
         access_key = self.get_access_key()
         if access_key is None:
+            logger.warning("S3 access key not provided. Defaulting to ''")
```
I do not see much point in this. If the access key is not provided, we should exit (throw an exception). Data Access factory do just this in validation

daw3rd replied:
done

cmadam replied:
If the S3 data access key is not provided, IMHO, the program should use the credentials given in the `~/.aws/credentials` file.  This is the default behavior of the `boto3` client.

blublinsky replied:
Sorry, no. We tried it before and it led to a lot of confusion. Also, remember, our main goal is running on the cluster not locally

blublinsky commented on data-processing-lib/python/src/data_processing/data_access/data_access_s3.py:
```diff
@@ -51,19 +51,17 @@ def __init__(
             access_key = ""
         secret_key = self.get_secret_key()
         if secret_key is None:
+            logger.warning("S3 secret key not provided. Defaulting to ''")
```
I do not see much point in this. If the secret key is not provided, we should exit (throw an exception). Data Access factory do just this in validation

daw3rd replied:
done

cmadam replied:
Is the data access factory currently checking the `~/.aws/credentials` file when running locally?

blublinsky commented on data-processing-lib/python/src/data_processing/data_access/data_access_s3.py:
```diff
@@ -73,17 +71,31 @@ def __init__(
 
     def get_region(self):
-        return (self.s3_credentials.get("region", None),)
+        return self.s3_credentials.get("region", None)
```
Sorry, these are completely artificials. It was fine before. I do not see the need for these

daw3rd replied:
they are need for spark to set the credentials in hadoop.

blublinsky commented on data-processing-lib/ray/pyproject.toml:
```diff
@@ -1,23 +1,17 @@
-    "mmh3",
-    # These two are to fix security issue
-    "fastapi>=0.109.1",
-    "pillow>=10.2.0",
```
Removed these 2 will create security issues in the image

daw3rd replied:
I removed them because I could not find them in the image or venv if i removed them here.  Maybe a specific transform is/was using them, but I can't see them any where now.

daw3rd replied:
I moved these from the python toml back to the ray toml.

blublinsky commented on data-processing-lib/ray/pyproject.toml:
```diff
@@ -1,23 +1,17 @@
     { name = "Boris Lublinsky", email = "blublinsky@ibm.com" },
 ]
 dependencies = [
+    "data-prep-toolkit",
```
I think we need a version here

daw3rd replied:
done

daw3rd committed:
Change DataAccesS3 to throw exception if now access or secret key are provided.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Add version number to ray and spark toml for data-prep-toolkit dependency

Also add makefile check on building spark images to make sure the base image exists.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Forget to change order of lib install for ingest2parquet.

Signed-off-by: David Wood <dawood@us.ibm.com>

daw3rd committed:
Also fix order of lib installation in kfp Dockerfile

Signed-off-by: David Wood <dawood@us.ibm.com>
